src/Compilers/Core/Portable/Diagnostic/DiagnosticSeverity.cs

```diff
@@ -27,7 +27,12 @@ public enum DiagnosticSeverity
         /// <summary>
         /// Something not allowed by the rules of the language or other authority.
         /// </summary>
-        Error = 3
+        Error = 3,
+
+        /// <summary>
+        /// Something that can be potentially improved.
+        /// </summary>
+        Suggestion = 4,
```

Comment:
> user puts warning and then for warnings that they provide code fix ... they can augment it with suggestion tag.

This feels like this would be super noisy.  A squiggle is already quite 'in your face'.  I don't think adding more adornments would be desirable.

>  then they will know which their diagnostic ID has fix and which one doesn't have)

You can already tell this by seeing that a lightbulb pops up when you go to that squiggle.

The issue is for diagnostics that are not warnings or errors, and thus do not have that existing squiggle treatment in the IDE.  There is no way to even know that there's a problem (esp. as we're likely to not show Suggestions by default in the diagnostic list).  

As such, like with warnings and errors, there is a desire to have some UI affordance to let the user know about the issue and to easily ascertain the *severity* of the issue.  Warning and Error already let you know what the severity it is by having a large squiggle and having that squiggle be red or green.  Neither of these approaches are ideal for suggestions as they're very much in your face.  We'd like something that ideally fades out a bit and isn't so imposing.